### PR TITLE
arrayfire 3.9.0

### PR DESCRIPTION
**Pull request by ZhongRuoyu**
Created by `brew bump`

---

Created with `brew bump-formula-pr`.

### CODE REVIEW

**Comment by ZhongRuoyu** [comment]
```
  In file included from /tmp/arrayfire-20230829-68291-1sdlwds/arrayfire-full-v3.9.0/build/extern/ocl_clfft-src/src/library/plan.cpp:29:
  /tmp/arrayfire-20230829-68291-1sdlwds/arrayfire-full-v3.9.0/build/extern/ocl_clfft-src/src/library/action.h:54:13: error: exception specification of overriding function is more lax than base version
      virtual ~FFTStockhamAction() = default;
              ^
  /tmp/arrayfire-20230829-68291-1sdlwds/arrayfire-full-v3.9.0/build/extern/ocl_clfft-src/src/library/../statTimer/../library/plan.h:369:13: note: overridden virtual function is here
      virtual ~FFTAction() = default;
              ^
  In file included from /tmp/arrayfire-20230829-68291-1sdlwds/arrayfire-full-v3.9.0/build/extern/ocl_clfft-src/src/library/plan.cpp:29:
  /tmp/arrayfire-20230829-68291-1sdlwds/arrayfire-full-v3.9.0/build/extern/ocl_clfft-src/src/library/action.h:182:13: error: exception specification of overriding function is more lax than base version
      virtual ~FFTGeneratedStockhamAction() = default;
              ^
  /tmp/arrayfire-20230829-68291-1sdlwds/arrayfire-full-v3.9.0/build/extern/ocl_clfft-src/src/library/action.h:54:13: note: overridden virtual function is here
      virtual ~FFTStockhamAction() = default;
              ^
```

**Comment by xyproto** [comment]
`pkg-config` appears to be a required dependency

**Comment by ZhongRuoyu** [comment]
> `pkg-config` appears to be a required dependency

It doesn't seem to be required. The problem on Linux is that OpenCL is missing.

**Comment by ZhongRuoyu** [comment]
[The failure on Linux](https://github.com/Homebrew/homebrew-core/actions/runs/6045197566/job/16404962224?pr=140833#step:4:5658) is strange:

```
  Error: An exception occurred within a child process:
    NoMethodError: undefined method `header' for nil:NilClass
```

File: Formula/a/arrayfire.rb [hunk 1]
```diff
@@ -42,11 +47,18 @@ def install
...
                     *std_cmake_args
     system "cmake", "--build", "build"
     system "cmake", "--install", "build"
 
+    # Remove debug info. These files make patchelf fail.
+    rm_f [
+      lib/"libaf.debug",
+      lib/"libafcpu.debug",
+      lib/"libafopencl.debug",
+    ]
```
Comment (ZhongRuoyu) [inline Formula/a/arrayfire.rb:61]: The Linux failure was with these debug info files, which patchelf.rb could not handle. Now we should be good.

**Review by fxcoudert — APPROVED** [review]

**Pull request merged**